examples/script_nifti.py

```diff
@@ -0,0 +1,41 @@
+import os
+import pathlib
+import shutil
+import sys
+from shimmingtoolbox.dicom_to_nifti import dicom_to_nifti
+import tempfile
```

Comment:
Can this have a better name than `script_nifti`? What is its overall purpose?

Your introduction comment on the PR says this is supposed to be an example, but I know it's filling in the "dicom_to_nifti" roll from https://github.com/shimming-toolbox/shimming-toolbox/blob/f320602c07c97216176daaee3ebf8ca0542368da/%2Bimutils/dicom_to_nifti.m. But reading the code so far, it's also got parts of

https://github.com/shimming-toolbox/shimming-toolbox/blob/9081cdac6701d7df74c5db109c205655be10d54e/example/experiment.m#L26-L30

Mostly all that function did was run [`dcm2bids`](https://github.com/cbedetti/Dcm2Bids) with some specialized options, which is itself already in python. If that's all we want, it seems like this should be a function named `shimmingtoolbox.util.dcm2bids()` that is just a thin wrapper to `dcm2bids`. And the part that downloads test data and exercises it? That should become a unit test:

```
# tests/test_dcm2bids.py

@pytest.mark.parameterize('dataset_url', ['https://github.com/shimming-toolbox/data-testing/archive/r20200713.zip'])
def test_dcm2bids(dataset_url):
    with urlopen(dataset_url) as ..., ... as ...:
        shimmingtoolbox.util.dcm2bids(.....)
```

Reply:
As an aside, there is a school of thought that says **all** your examples should be unit tests. If someone wants to learn how to use your software they should read that tests, and that the tests should be good enough to teach how to use your software.

I think they're too hardline; good readable examples help get people into projects, and unit tests don't fit the same niche. But it is true that pure example code is not code that's getting exercised regularly.

Reply:
While I'm thinking about it, pytest even has [this whole system](https://docs.pytest.org/en/stable/fixture.html) that would let us abstract out that dataset downloading thing nicely:

```
@pytest.mark.parameterize('dataset_url', ['https://github.com/shimming-toolbox/data-testing/archive/r20200713.zip'])
def test_dcm2bids(downloaded_data):
    shimmingtoolbox.util.dcm2bids(downloaded_data.path, ...)
```